### DISCUSSION TITLE
New Plugins endpoint not Working

### DISCUSSION THREAD

**Discussion by Carnotaurus454** [danny-avila/LibreChat#533] [Troubleshooting] [answered]
With no plugins enabled the plugins endpoint still returns that I need to set my openAI API key. I have it set and the normal openAI endpoint works fine. Although, I'm not getting the 'error to authenticate this plugin', it lets me install the plugins fine.

I did a little testing. I have the project setup on my raspberry pi (it allows me to use gpt at school cos its blocked :/ and I want to use browsing ect. I can also teach my friends how to properly use it lol, really sucks its blocked cos people need to learn this) The pi version returned the invalid API key error. I got gpt to write some python to validate my CSE and Google API key and it all works. I've tried setting it up locally on docker as well and that returns the same. Is there anything that could be causing this that would come to mind? I've got to say this project is amazing and I love all your work. :)

The exact error is: Error: Failed to send message. HTTP 401 - {
    "error": {
        "message": "",
        "type": "invalid_request_error",
        "param": null,
        "code": "invalid_api_key"
    }
}
    at onopen (/app/api/app/langchain/ChatAgent.js:305:25)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async create (/app/node_modules/@waylaidwanderer/fetch-event-source/lib/cjs/fetch.cjs:56:17) {
  status: 401,
  json: {
    error: {
      message: '',
      type: 'invalid_request_error',
      param: null,
      code: 'invalid_api_key'
    }
  }
}

**Comment by danny-avila** [comment]
Hey, thanks so much! Glad it could help you! Are you setting your OPENAI_API_KEY in the .env file ? Do you mind sharing your env variables (withoutprivate credentials) ?

I just tried it on a fresh install and I only have issues when it's actually not a valid API KEY

![image](https://github.com/danny-avila/LibreChat/assets/110412045/fa0e85e8-5e39-4b25-8410-0162e4155dde)

**Setting my OPEN_API_KEY works**

![image](https://github.com/danny-avila/LibreChat/assets/110412045/a2351d59-84ce-4162-b52b-33a2b3e9579f)

**Reply by Carnotaurus454** [reply, marked as answer]
I have the OpenAI Api key as "user provided" in the .env to allow for my mates to use it without using my key. I'll give it a test with my key in the env. I have just been setting it in the web interface.

**Reply by Carnotaurus454** [reply]
thanks heaps

**Comment by Carnotaurus454** [comment]
yayyayayy it works nowwww. Just wondering, could you allow the token for plugins to be user set in the web interface?
